### ISSUE TITLE
USWDS - Bug: The footer's contact info links aren't a11y-compliant

### ISSUE THREAD

**Issue by andycochran** [uswds/uswds#5557]
### Describe the bug

A 508 compliance review for our project revealed that the contact info links in the footer — `.usa-footer__contact-info a` — are non-compliant. Links are expected to be underlined _before_ hovering over with a mouse. There is no visual distinction between these links and regular text. Not even color. 

### Steps to reproduce the bug

View the footer before hovering the links: 

![image](https://github.com/uswds/uswds/assets/409279/64a6a644-7499-422b-ab14-a18654b22368)

…and after hovering the links (left one): 

![image](https://github.com/uswds/uswds/assets/409279/4b0da07b-1d43-41bb-9f01-1b431c47fed4)

It is not visually apparent that the texts are links until you hover. 

### Expected Behavior

The contact info links in the footer are underlined before hovering them. 

### Screenshots

(see above) 

### System setup

n/a (not system-specific)

### Additional context

See also https://github.com/HHS/grants-equity/issues/563 

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://designsystem.digital.gov/about/community/#community-conduct).
- [X] I checked the [current issues](https://github.com/uswds/uswds/issues?q=is%3Aissue+is%3Aopen+label%3A%22Type%3A+Bug%22) for duplicate bug reports.

**Comment by mahoneycm** [comment]
Confirming that current footer link styles do not meet WCAG 2.0 AA standards. If a link does not have an underline in its default state, it must have sufficient color contrast from surrounding text.

> The link text must have a 3:1 contrast ratio from the surrounding non-link text.
>
> The link must present a "non-color designator" (typically the introduction of the underline) on both mouse hover and keyboard focus.
>
> WCAG recognizes that meeting these contrast requirements “is not the preferred technique to differentiate link text”. Of course the easy solution to this is to simply underline links in their default state. Interestingly, underlined links can be the exact same color as their surrounding text, though this is far from optimal. [^1]

The most straightforward route would be to add the underline during default states. Alternatively, we could increase color contrast between the link text and the surrounding footer text.

[^1]: https://webaim.org/blog/wcag-2-0-and-link-colors/
